### ISSUE TITLE
Some feature requests

### ISSUE THREAD

**Issue by eXmendiC** [yaser01/mkv-muxing-batch-gui#111]
First of all, really great tool.
I did something similar with the CLI, but it was always kinda annoying to modify it everytime to my needs (sometimes multiple audio, sometimes multiple subs etc.). However, there are a few features that would definitly improve this tool.

1) Select new "video" (input) either as new videos/input (the way it's now) or add for an existing input
Kinda like a dialoge box, that asks what you want to keep from "Source 1" and what from "Source 2". 
That will fill the entries subtitles, audios, chapters, attachments for you. For now, I've to demux "Source 2" beforehand and select it as new input for every entry. Something like that is already possible with MKVtoolnix, where you can easily decide what you want to keep from a second, third, etc. input. However, MKVtoolnix lacks a batch ability, which overall isn't worth that. This is really useful if I just want to replace a video with better quality, but want to keep everything else or want to combine already muxed multiple language sources to one. Since it would autofill audios, subtitles and chapters/attachments for you, it's still possible for me to add delay for some of them.

2) Delay for chapters
mkvmerge allows to add delay to chapters. This is useful, if you replace the video and it doesn't match with the rest anymore, but the old one included chapters while the new one doesn't. However, what mkvmerge lacks is the abbility to have positive delay without creating a new unnecessary chapter between it and 00:00 (video start). That's because if I shift everything with for example +1001, it will also shift the first chapter. Maybe there is a way to avoid that and hardcode the first chapter to 00:00:00. 

3) Allow renaming the output (with variables)
Maybe this is actually a thing, but in the "Mux settings" tab I couldn't find an option for that.
Example:
[Releasetag] Name - %episode% [%sublang%][BD %resolution% %audiocodec%][%crc%].mkv
Some example variables are probably "episode" (which would be just an counter, starting with the value you want), "audiocodec" (which does already exist in the "Audios" tab to select) and "crc". Maybe more is variables are possible, if it's possible to read them via media info.

4) Add other options that exist in mkvmerge (MkvToolNix) which I consider useful
--stop-after-video-ends - Enabled: If the video is longer than the audio (which could happen with a remux), this wouldn't be the case anymore
--remove-dialog-normalization-gain - Useful for AC3/EAC3 & DTS tracks
--cropping - Useful, because newer video players actually respect that value now
--fix-bitstream-timing-information - Enabled, the video still starts at 00:00:00 even if the audio codec has an internal delay

But maybe instead of adding all these, an option to add custom CLI settings would be more useful (and maybe a custom mkvmerge version).

**Comment by yaser01** [comment]
First of all, you are awesome :D
The effort with writing all of this is appreciated...a lot :D
1. Massive feature to be honest...won't drop it.. but need a lot of thinking to make it look as best as possible.
2. Finally some new feature to the chapter tab :D, the chapter tab never have new update since it's added, just want more explaining for the "hardcode delay thing" can you give me example with maybe 3-4 chapters with their start and end, and what mkvtoolnix default doing.. and what is **_better_** as you described.
3. it's old request by some people will work on it for sure, but it won't be using media info.. it has to be named by the user...
4. someone already proposed this to be a thing .. but i didn't know what to add.. i will probably this of a way to make the user add whatever he wants and verified that the command exist before muxing everything...
Thanks a lot again, brilliant ideas that needs a lot of work for a person with _no job yet_ :D
Have a nice day.

**Comment by eXmendiC** [comment]
Yeah, some of the features are probably demanding a lot of time. After all, these are just requests/suggestions to improve the tool.

Here is an example for what I mean with chapters:
[Test_Chapters.txt](https://github.com/yaser01/mkv-muxing-batch-gui/files/13924679/Test_Chapters.txt)
As you can see, the first chapter starts with 0. That's how it has to be.
`<ChapterTimeStart>00:00:00.000000000</ChapterTimeStart>`

Let's say the difference between the old source (including these chapters) and new source are 24 black frames at the beginning of the new source, which is a pretty common scenario (at least for me). Thus, I have to add a delay to all of these chapters, which is 1001ms (for 23,976fps):
![mkvtoolnix](https://github.com/yaser01/mkv-muxing-batch-gui/assets/26603887/4dd91e6c-f7a8-473a-9cc8-f35cf8741014)
If you want to try that out, any video works fine muxing. Only the chapter file matters for testing.

So far so good, but as soon as you extract the chapters file with the new muxed version, you notice that the first chapter doesn't start at 0 anymore:
`<ChapterTimeStart>00:00:01.001000000</ChapterTimeStart>`
[Test_Chapters_New.txt](https://github.com/yaser01/mkv-muxing-batch-gui/files/13924761/Test_Chapters_New.txt)

Video players (like mpv) add an invisible chapter "0" between 00:00:00.000000000 and 00:00:01.001000000, something you don't really want here. They should start with chapter "1". So in case you press the chapter skipping button or use Plex for that, you don't end up at the "Opening" (like you would if the first chapters starts with 0) and end up with "Prologue" instead. Kinda annoying.
![mpv](https://github.com/yaser01/mkv-muxing-batch-gui/assets/26603887/8d191399-4162-43ba-bd90-8b975e6b3a37)

So far my workaround is to manually change that single line after extracting the new muxed version and doing a second muxing with the new "fixed version" of these chapters again. A step that could be avoided, if there is more logic behind the delaying process - like not touching anything with "00:00:00.000000000".
